Heroic-Games-Launcher/HeroicGamesLauncher
# [Feat/macOS] Add Support for Apple's Gaming Toolkit

With this, Heroic will look for a current setup of the apple's GPT and list it under the Wine List:

So far it works for Epic games, GOG and also Sideloaded games.

Testers Needed!


## HOW TO TEST: EPIC and GOG Games

- To download the build to test, go bellow under 'CHECKS' and build_macOS click on `details`. Then on the page that will open, click on SUMMARY. And then go to ARTIFACTS and download the build for your system.
- Since this build is NOT signed (only releases are), you need to first run this command on the terminal: `xattr -d -r com.apple.quarantine PATH_TO/Heroic.app`
- Make sure you have installed the gaming porting toolkit following the Install instructions provided by Apple.
- Make sure Spotlight is enabled and the command `mdfind wine64` returns something on the terminal.
- **[IMPORTANT] If you want to use the Apple scripts, copy the 3 scripts (gamingportingtoolkit...) to `$HOME/bin` folder, otherwise Heroic won't be able to find them.**
- Try to run some games and make sure it is working already.
- When opening Heroic, go to Settings and on Game default check if, under WINE, it shows the list of the 3 gaming toolkit scripts as below and well the standalone wine (wine-7.7):
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/26871415/5ff90661-7eda-4f77-ad67-8ad0e2a04fb0)

- Download and install some Windows games from EPIC or GOG and select the Script you want to use on the game wine settings.
- Use new prefixes instead of reusing an existing ones. If you have DXVK installed on the prefix, the game might not work.

## Sideloaded Games

- Tested only Steam so far
- Click on ADD GAMES on the library.
- Fill in the title as you want.
- Select the Gaming Toolkit script you want to use.
- Click on RUN INSTALLER FIRST and select the exe you want to install
- The app will be installed inside the prefix
- On Select EXE, select the exe to run the app, for instance: `/Users/name/Games/GPT_PREFIX/drive_c/Program Files (x86)/Steam/Steam.exe`
- Click Finish

---

Use the following Checklist if you have changed something on the Backend or Frontend:
